docs/build/interoperate-qiskit-qasm2.mdx

```diff
@@ -51,63 +59,179 @@ import qiskit.qasm2
 circuit = qiskit.qasm2.load("myfile.qasm")
 ```
 
+
 <span id="custom-instructions"></span>
-## Custom quantum instructions
+### Link OpenQASM 2 gates with Qiskit gates
+
+By default, Qiskit's OpenQASM 2 importer treats the include file `"qelib1.inc"` as a *de facto* standard library.
+The importer treats this file as containing precisely the gates it is described to contain in [the original paper defining OpenQASM 2](https://arxiv.org/abs/1707.03429).
+Qiskit will use the built-in gates in [the circuit library](../api/qiskit/circuit_library) to represent the gates in `"qelib1.inc"`.
+Gates defined in the program by manual OpenQASM 2 `gate` statements will, by default, be constructed as custom [Qiskit `Gate` subclasses](../api/qiskit/qiskit.circuit.Gate).
+
+You can tell the importer to use specific [`Gate`](../api/qiskit/qiskit.circuit.Gate) classes for the given `gate` statements it encounters.
+You can also use this mechanism to treat additional gate names as "built-in", that is, not requiring an explicit definition.
+Qiskit will typically be more efficient if you can tell it which gate classes it should use for `gate` statements outside of `"qelib1.inc"`.
+
+<Admonition type="warning">
+As of Qiskit SDK 1.0, Qiskit's OpenQASM 2 *exporter* (see [Export a Qiskit circuit to OpenQASM 2](#export)) still behaves as if `"qelib1.inc"` has more gates than it really does.
+This means that the default settings of the importer might not be able to import a program exported by our importer.
+See [the specific example on working with the legacy exporter](#import-legacy) to resolve this problem.
+
+This discrepancy is legacy behavior of Qiskit, and [we are planning to fix it in a later release of Qiskit](https://github.com/Qiskit/qiskit/issues/10737).
```

Comment:
```suggestion
This discrepancy is legacy behavior of Qiskit, and [it will be resolved in a later release of Qiskit](https://github.com/Qiskit/qiskit/issues/10737).
```